Handle new GwcsBoundingBoxWarning

```
_____________________________ test_jwst_s3d_single _____________________________
...
>       data = Spectrum1D.read(tmpfile, format='JWST s3d')

../../specutils/io/default_loaders/tests/test_jwst_reader.py:435: 
...
../../specutils/io/default_loaders/jwst_reader.py:590: in _jwst_s3d_loader
    grid = grid_from_bounding_box(wcs.bounding_box)[:, :, 0, 0]
...
E           gwcs.wcs.GwcsBoundingBoxWarning: The bounding_box was set in C order on the transform prior to being used in the gwcs!
E           Check that you indended that ordering for the bounding_box, and consider setting it in F order.
E           The bounding_box will remain meaning the same but will be converted to F order for consistency in the GWCS.
```

Probably can ignore like https://github.com/spacetelescope/jdaviz/pull/3283 but should double check if you are using the bounding box in C- or F-order first. See https://github.com/spacetelescope/gwcs/pull/522

--- a/specutils/io/default_loaders/jwst_reader.py
+++ b/specutils/io/default_loaders/jwst_reader.py
@@ -479,7 +479,11 @@ def _jwst_s2d_loader(filename, **kwargs):
 
             # Get the wavelength array from the GWCS object which returns a
             # tuple of (RA, Dec, lambda)
-            grid = grid_from_bounding_box(wcs.bounding_box)
+            with warnings.catch_warnings():
+                # https://github.com/spacetelescope/gwcs/pull/522
+                warnings.filterwarnings(
+                    "ignore", message="The bounding_box was set in C order.*")
+                grid = grid_from_bounding_box(wcs.bounding_box)
             _, _, lam = wcs(*grid)
             _, _, lam_unit = wcs.output_frame.unit
 
@@ -587,7 +591,11 @@ def _jwst_s3d_loader(filename, **kwargs):
             # tuple of (RA, Dec, lambda).
             # Since the spatial and spectral axes are orthogonal in s3d data,
             # it is much faster to compute a slice down the spectral axis.
-            grid = grid_from_bounding_box(wcs.bounding_box)[:, :, 0, 0]
+            with warnings.catch_warnings():
+                # https://github.com/spacetelescope/gwcs/pull/522
+                warnings.filterwarnings(
+                    "ignore", message="The bounding_box was set in C order.*")
+                grid = grid_from_bounding_box(wcs.bounding_box)[:, :, 0, 0]
             _, _, wavelength_array = wcs(*grid)
             _, _, wavelength_unit = wcs.output_frame.unit
 

